tests/testthat/test-0diversity.R

```diff
@@ -38,4 +38,19 @@ test_that("diversity estimates", {
     expect_equal(unname(round(cd$coverage, 0)), c(2,3,1))
     expect_equal(unname(round(cd$fisher, 4)), c(8.8037, 10.0989, 13.2783))
 
+    # Tests faith index with esophagus data
+    for( i in c(1:(length(colnames(tse_idx)))) ){
+        # Gets those taxa that are present/absent in the sample
+        present <- rownames(tse)[assays(tse)$counts[,i] > 0]
+        absent <- rownames(tse)[assays(tse)$counts[,i] == 0]
+
+        # Absent taxa are dropped from the tree
+        sub_tree <- ape::drop.tip(rowTree(tse_idx), absent)
+        # Faith is now calculated based on the sub tree
+        faith <- sum(sub_tree$edge.length)
+
+        expect_equal(cd$faith[i], faith)
+    }
+
+
```

Comment:
Can you add test for all the combination mentioned above?
- SE + tree
- TSE 
- TSE + tree

hint: you can always "downgrade" a TSE to SE using `as`.

Reply:
Added